src/core/raster/qgssinglebandpseudocolorrenderer.cpp

```diff
@@ -38,7 +38,7 @@ QgsSingleBandPseudoColorRenderer::QgsSingleBandPseudoColorRenderer( QgsRasterInt
 
 void QgsSingleBandPseudoColorRenderer::setBand( int bandNo )
 {
-  if ( bandNo > mInput->bandCount() || bandNo <= 0 )
+  if ( !mInput || bandNo > mInput->bandCount() || bandNo <= 0 )
```

Comment:
actually in this case we should skip the check, and just set the band number directly. Otherwise it breaks the workflow of invalid layers

Reply:
True.